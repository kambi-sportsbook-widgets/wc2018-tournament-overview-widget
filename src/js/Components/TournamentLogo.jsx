import React from 'react'
import PropTypes from 'prop-types'
import { translationModule } from 'kambi-widget-core-library'

import styles from './TournamentLogo.scss'
import mobile from '../Services/mobile'

const t = translationModule.getTranslation.bind(translationModule)

const TournamentLogo = ({ logoUrl, scrolledListHasHorizontalSpaceLeft }) => {
  let className = styles.general
  if (scrolledListHasHorizontalSpaceLeft) {
    className = `${className} ${styles.scrolledListHasHorizontalSpaceLeft}`
  } else if (!mobile()) {
    className = `${className} ${styles.desktop}`
  }
  return (
    <div className={className}>
      <i
        className="kw-custom-logo-large-type"
        style={{
          backgroundImage: `url("${logoUrl}")`,
        }}
      />
      <div className={styles.tournament}>
        <p>{t('wc2018').toUpperCase()}</p>
        <p>{t('russia').toUpperCase()}</p>
      </div>
    </div>
  )
}

TournamentLogo.propTypes = {
  /**
   * Logo CSS class name.
   * These classes are defined in operator-specific CSS file.
   */
  logoName: PropTypes.string,
}

TournamentLogo.defaultProps = {
  logoName: 'football',
}

export default TournamentLogo
