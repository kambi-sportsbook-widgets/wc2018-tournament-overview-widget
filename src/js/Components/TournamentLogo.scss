.general {
   display: flex;
   align-items: center;
   height: 120px;
   flex-wrap: wrap;
   overflow: visible;
   z-index: 1;
   max-width: 150px;

   i {
      background: no-repeat 50%;
      background-size: contain;
      height: 80px;
      width: 80px;
      min-width: 80px;
      flex-basis: 100%;
   }

   &.scrolledListHasHorizontalSpaceLeft {
      width: 0;
      max-width: 0;
      direction: rtl;
   }


   &.desktop {
      height: 100%;
      align-self: center;
      display: flex;
      flex-direction: row;
      text-align: center;
      justify-content: center;
      max-width: 180px;

      i {
         flex-basis: 100%
      }
   }

   .tournament {
      flex-basis: 100%;
      text-align: center;
      p {
         margin: 0;
         line-height: 1.4;

         &:first-child {
            font-weight: 600;
            font-size: 1.2rem;
            margin-top: .5rem;
            letter-spacing: .7px;
         }
      }         
   }
}
