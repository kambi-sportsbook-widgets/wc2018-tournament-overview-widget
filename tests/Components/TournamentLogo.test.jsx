/* eslint-env jest */
import React from 'react'
import ReactShallowRenderer from 'react-test-renderer/shallow'
import { translationModule } from 'kambi-widget-core-library'

import TournamentLogo from '../../src/js/Components/TournamentLogo'

let renderer
let mockMobile = false
jest.mock('../../src/js/Services/mobile', () => () => mockMobile)
jest.mock('kambi-widget-core-library', () => ({
  translationModule: {
    getTranslation: jest.fn(key => `Translated ${key}`),
  },
}))

describe('Event DOM rendering', () => {
  beforeEach(() => {
    renderer = new ReactShallowRenderer()
    mockMobile = false
    translationModule.getTranslation = jest.fn()
  })

  it('renders correctly with default props', () => {
    expect(renderer.render(<TournamentLogo />)).toMatchSnapshot()
  })

  it('renders correctly with custom logo CSS class name', () => {
    expect(
      renderer.render(<TournamentLogo logoClassName="test_logo_class_name" />)
    ).toMatchSnapshot()
  })
})
