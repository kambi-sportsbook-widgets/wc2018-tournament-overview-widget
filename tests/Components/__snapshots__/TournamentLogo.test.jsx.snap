// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event DOM rendering renders correctly with custom logo CSS class name 1`] = `
<div
  className="general desktop"
>
  <i
    className="kw-custom-logo-large-type"
    style={
      Object {
        "backgroundImage": "url(\\"undefined\\")",
      }
    }
  />
  <div
    className="tournament"
  >
    <p>
      TRANSLATED WC2018
    </p>
    <p>
      TRANSLATED RUSSIA
    </p>
  </div>
</div>
`;

exports[`Event DOM rendering renders correctly with default props 1`] = `
<div
  className="general desktop"
>
  <i
    className="kw-custom-logo-large-type"
    style={
      Object {
        "backgroundImage": "url(\\"undefined\\")",
      }
    }
  />
  <div
    className="tournament"
  >
    <p>
      TRANSLATED WC2018
    </p>
    <p>
      TRANSLATED RUSSIA
    </p>
  </div>
</div>
`;
